script.js

```diff
@@ -1,4 +1,82 @@
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  var opponentSelection = scissorsPaperStone();
+  var lowerCaseOpponentSelection = opponentSelection.toLowerCase();
+  var lowerCaseInput = input.toLowerCase();
+  var myOutputValue =
+    "Sorry, you lost! You selected " +
+    input +
+    " but the opponent selected " +
+    opponentSelection +
+    ". Please try again!";
+  // You win under the following conditions:
+  // 1. You select Scissors, and opponent select Paper,
+  // 2. OR you select Paper, and opponent select Stone,
+  // 3. OR you select Stone, and opponent select Scissors.
+  if (
+    (lowerCaseInput == "scissors" && lowerCaseOpponentSelection == "paper") ||
+    (lowerCaseInput == "paper" && lowerCaseOpponentSelection == "stone") ||
+    (lowerCaseInput == "stone" && lowerCaseOpponentSelection == "scissors")
+  ) {
+    myOutputValue =
+      "Congratulations, you win! You selected " +
+      input +
+      " while the opponent selected " +
+      opponentSelection +
+      ".";
+  }
+  // It will be a draw under the following conditions:
+  // 1. You select Scissors, and opponent select Scissors,
+  // 2. OR you select Paper, and opponent select Paper,
+  // 3. OR you select Stone, and opponent select Stone.
+  if (
+    (lowerCaseInput == "scissors" &&
+      lowerCaseOpponentSelection == "scissors") ||
+    (lowerCaseInput == "paper" && lowerCaseOpponentSelection == "paper") ||
+    (lowerCaseInput == "stone" && lowerCaseOpponentSelection == "stone")
+  ) {
+    myOutputValue = "It's a draw!";
+  }
+  // If user types anything other than a valid choice, he/she will be asked to try again.
+  if (
+    !(
+      lowerCaseInput == "scissors" ||
+      lowerCaseInput == "paper" ||
+      lowerCaseInput == "stone"
+    )
+  ) {
+    myOutputValue =
+      "Sorry, " +
+      input +
+      " is not a valid choice. Please input only scissors, paper or stone.";
+  }
   return myOutputValue;
 };
+
+var opponentRNG = function () {
+  // This function will randomly generate numbers 1 to 3.
+  // Generate a random number from 0 to 2.9999 recurring.
+  var randomNumber = Math.random() * 3;
+
+  // Generate a random integer from 0 to 2, by removing the decimals.
+  var randomInteger = Math.floor(randomNumber);
+
+  // Add 1 to generate a random integer from 1 to 3.
+  var oneToThree = randomInteger + 1;
+
+  return oneToThree;
+};
+
+var scissorsPaperStone = function () {
+  // The default result is Stone, if RNG is 3 (i.e. 3 = Stone).
+  var opponentRandomNumber = opponentRNG();
+  var opponentSPS = "Stone";
+  // If RNG = 2, result is Paper (i.e. 2 = Paper).
+  if (opponentRandomNumber == 2) {
+    opponentSPS = "Paper";
+  }
+  // If RNG = 1, result is Scissors (i.e. 1 = Scissors)
+  if (opponentRandomNumber == 1) {
+    opponentSPS = "Scissors";
+  }
```

Comment:
I noticed here that you are storing the values of the opponent's selection as strings with the first letter capitalised, like `"Stone"` instead of `"stone"`.

If you have stored the opponent's selection string using lower-cased string values right away, like `"stone"`, [line 3](https://github.com/rocktheship/basics-scissors-paper-stone/blob/3db4d589a89fe56981c6bfd49cb0bcbfc6fe0f63/script.js#L3) will not be necessary anymore. There will be no need to even lower-case the opponent's string values, because they are already in lower-case.

Doing this also cascades to simplifying your other `if` conditions just a little bit. In [lines 15 to 19](https://github.com/rocktheship/basics-scissors-paper-stone/blob/3db4d589a89fe56981c6bfd49cb0bcbfc6fe0f63/script.js#L15-L19), you need not use `lowerCaseOpponentSelection` anymore. You can simply use `opponentSelection` instead, like so:

```
// opponentSelection is already in lower-case
// thus, you can directly compare the lower-case opponentSelection
// with the lower-case string value in your conditionals here
// like "paper"
if (
    (lowerCaseInput == "scissors" && opponentSelection == "paper") ||
    (lowerCaseInput == "paper" && opponentSelection== "stone") ||
    (lowerCaseInput == "stone" && opponentSelection == "scissors")
  ) {
    myOutputValue =
      "Congratulations, you win! You selected " +
      input +
      " while the opponent selected " +
      opponentSelection +
      ".";
  }
```